src/features/users/components/chef-follow-button.tsx

```diff
@@ -1,18 +1,45 @@
 "use client";
 
-import { useState } from "react";
+import { experimental_useOptimistic as useOptimistic } from "react";
+import { useRouter } from "next/navigation";
 
-import { Button } from "@/components/button/button";
+import { updateUserFollow } from "@/features/users";
 
-export const ChefFollowButton = () => {
-  const [isFollow, setFollow] = useState(false);
-  const toggleFollow = () => setFollow(!isFollow);
+export function ChefFollowButton({
+  initialIsFollowed,
+  chefId,
+  loginUserId,
+}: {
+  initialIsFollowed?: boolean;
+  chefId: string;
+  loginUserId?: string;
+}) {
+  const router = useRouter();
+  const [optimisticIsFollowed, optimisticSetIsFollowed] = useOptimistic(
```

Comment:
(修正必須じゃないです)

好みだと思いますが、配列の2つめは関数だと思うので
動詞から始まる方がいいかと思いました
公式も `add~`って感じですし \
https://nextjs.org/docs/app/building-your-application/data-fetching/forms-and-mutations#optimistic-updates


```suggestion
  const [optimisticIsFollowed, setOptimisticIsFollowed] = useOptimistic(
```

Reply:
@daiki-kon 
ありがとうございます、下記コミットで対応しました
[5d0ea02](https://github.com/qin-team-recipe/08-recipe-app/pull/169/commits/5d0ea02ee8b2b96874c108c1f1cc0091201da130)